### PR TITLE
feat(vitals): Support measurements in metric alerts

### PR DESCRIPTION
**Pull request by dashed**
Expose web vitals as options in metric alerts.

Depends on https://github.com/getsentry/sentry/pull/21566

## TODO

- [x] add test for measurement name validation

### CODE REVIEW

File: src/sentry/incidents/logic.py [hunk 1]
```diff
@@ -1350,7 +1350,7 @@ def get_column_from_aggregate(aggregate):
 
 def check_aggregate_column_support(aggregate):
     column = get_column_from_aggregate(aggregate)
-    return column is None or column in SUPPORTED_COLUMNS or column in TRANSLATABLE_COLUMNS
+    return column is None or column.startswith("measurements.") or column in SUPPORTED_COLUMNS or column in TRANSLATABLE_COLUMNS
```
Comment (Zylphrex) [inline src/sentry/incidents/logic.py:1353]: This should use `is_measurement` from `utils/snuba.py`

File: src/sentry/static/sentry/app/views/settings/incidentRules/metricField.tsx [hunk 2]
```diff
@@ -41,6 +42,12 @@ const getFieldOptionConfig = (dataset: Dataset) => {
         return ['tags[sentry:user]', 'string'];
       }
 
+      if (organization.features.includes('measurements')) {
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/views/settings/incidentRules/metricField.tsx:45]: Not sure how alerts handles this but I'd lean towards splitting measurements out from fields as they may require special handling as we see here. So maybe another list for just `measurements` in the config.

File: src/sentry/static/sentry/app/views/settings/incidentRules/metricField.tsx [hunk 3]
```diff
@@ -41,6 +42,12 @@ const getFieldOptionConfig = (dataset: Dataset) => {
...
+        if (key.startsWith('measurements.')) {
+          return [key, measurementType(key)];
+        }
+      }
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/incidentRules/metricField.tsx:49]: Do you also need to remove measurement fields if the organization doesn't have measures? It looks like line 51 will return the measure with undefined as a type.

File: src/sentry/static/sentry/app/views/settings/incidentRules/metricField.tsx (hunk 2 above)
Comment (Zylphrex) [reply src/sentry/static/sentry/app/views/settings/incidentRules/metricField.tsx:45]: As an aside, `generateFieldOptions` accepts a list of `measurementKeys` along side `fields`, so you can make use of that.

**Comment by wedamija** [comment]
Backend lgtm, just need review on the fe and we'll be good to merge

File: src/sentry/static/sentry/app/views/settings/incidentRules/metricField.tsx (hunk 2 above)
Comment (dashed) [reply src/sentry/static/sentry/app/views/settings/incidentRules/metricField.tsx]: Should be addressed in https://github.com/getsentry/sentry/pull/21498/commits/524d080acc0dbde1646fcd66d091a5ab49412ff0

**Review by markstory — APPROVED** [review]
Frontend bits look good to me.

**Pull request merged**